src/Integration.Vsix/SonarLintTracker/IssuesSnapshot.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace SonarLintTracker
+{
+    class IssuesSnapshot : WpfTableEntriesSnapshotBase
+    {
+        private readonly string filePath;
+        private readonly int versionNumber;
+
+        // We're not using an immutable list here but we cannot modify the list in any way once we've published the snapshot.
+        public readonly List<IssueMarker> IssueMarkers = new List<IssueMarker>();
```

Comment:
Property here. Generally from API POV, if you don't intend to let the users of this class to modify the collection the best practice is:

    private readonly List<IssueMarker> issueMarkers = new List<IssueMarker>();
    private readonly ReadOnlyCollection<IssueMarker> readonlyIssueMarkers = new ReadOnlyCollection<IssueMarker>(issueMarkers); // read-only wrapper to prevent casting the value of the following property to List<T>

    public IEnumerable<IssueMarker> IssueMarkers => readonlyIssueMarkers; // read-only property

    public void AddIssueMarker(IssueMarker marker) {... add to issueMarkers, throw exception if called after the snapshot was published ...}